adapters/dmx/dmx.go

```diff
@@ -0,0 +1,259 @@
+}
+
+type dmxExt struct {
+	Bidder dmxParams `json:"bidder"`
+}
+
+type dmxParams struct {
+	TagId       string `json:"tagid,omitempty"`
+	DmxId       string `json:"dmxid,omitempty"`
+	MemberId    string `json:"memberid,omitempty"`
+	PublisherId string `json:"publisher_id,omitempty"`
+	SellerId    string `json:"seller_id,omitempty"`
+}
+
+func UserSellerOrPubId(str1, str2 string) string {
+	if str1 != "" {
+		return str1
+	}
+	return str2
+}
+
+func (adapter *DmxAdapter) MakeRequests(request *openrtb.BidRequest, req *adapters.ExtraRequestInfo) (reqsBidder []*adapters.RequestData, errs []error) {
+	var dmxBidderStaticInfo dmxExt
+	var imps []openrtb.Imp
+	var rootExtInfo dmxExt
+	var publisherId string
+	var sellerId string
+
+	var dmxReq = request
+
+	if len(request.Imp) >= 1 {
+		err := json.Unmarshal(request.Imp[0].Ext, &rootExtInfo)
+		if err != nil {
+			errs = append(errs, err)
+		} else {
+			publisherId = UserSellerOrPubId(rootExtInfo.Bidder.PublisherId, rootExtInfo.Bidder.MemberId)
+			sellerId = rootExtInfo.Bidder.SellerId
+		}
+	}
+	if err := json.Unmarshal(request.Ext, &dmxBidderStaticInfo); err != nil {
+		errs = append(errs, err)
+	}
+
+	if dmxBidderStaticInfo.Bidder.PublisherId != "" && publisherId == "" {
+		publisherId = dmxBidderStaticInfo.Bidder.PublisherId
+	}
+
+	if dmxReq.App != nil {
+		dmxReq.Site = nil
+		dmxReq.App.Publisher.ID = publisherId
+	}
+	if dmxReq.Site != nil {
+		if dmxReq.Site.Publisher == nil {
+			dmxReq.Site.Publisher = &openrtb.Publisher{ID: publisherId}
+		} else {
+			if dmxReq.Site.Publisher != nil {
+				dmxReq.Site.Publisher.ID = adapter.publisherId
+			} else {
+				dmxReq.Site.Publisher.ID = publisherId
+			}
+		}
+	}
+
+	for _, inst := range request.Imp {
+		var banner openrtb.Banner
+		var ins openrtb.Imp
+
+		if len(inst.Banner.Format) != 0 {
+			banner = openrtb.Banner{
+				W:      &inst.Banner.Format[0].W,
+				H:      &inst.Banner.Format[0].H,
+				Format: inst.Banner.Format,
+			}
+
+			const intVal int8 = 1
+			var params dmxExt
+
+			source := (*json.RawMessage)(&inst.Ext)
+			if err := json.Unmarshal(*source, &params); err != nil {
+				errs = append(errs, err)
+			}
+			if params.Bidder.PublisherId == "" && params.Bidder.MemberId == "" {
+				var failedParams dmxParams
+				if err := json.Unmarshal(inst.Ext, &failedParams); err != nil {
+					errs = append(errs, err)
+					return nil, errs
+				}
+				imps = fetchAlternativeParams(failedParams, inst, ins, imps, banner, intVal)
+			} else {
+				imps = fetchParams(params, inst, ins, imps, banner, intVal)
+			}
+
+		}
+
+	}
+
+	dmxReq.Imp = imps
+
+	oJson, err := json.Marshal(request)
+
+	if err != nil {
+		errs = append(errs, err)
+	}
+	headers := http.Header{}
+	headers.Add("Content-Type", "Application/json;charset=utf-8")
+	reqBidder := &adapters.RequestData{
+		Method:  "POST",
+		Uri:     adapter.endpoint + addParams(sellerId), //adapter.endpoint,
+		Body:    oJson,
+		Headers: headers,
+	}
+
+	if dmxReq.User == nil {
```

Comment:
Consider moving this logic earlier in the method. A lot of work can be avoided if this is detected earlier.